src/provider/states/pod/running.rs

```diff
@@ -66,7 +64,7 @@ impl State<PodState> for Running {
         // Interruption of this loop is triggered externally by the Krustlet code when
         //   - the pod which this state machine refers to gets deleted
         //   - Krustlet shuts down
-        while !running_containers.is_empty() && !container_failed {
+        while !running_containers.is_empty() {
```

Comment:
I don't think we fully support multiple containers per pod yet, but in case we do that, my understanding is that this would consider the pod "running" until the last container has exited, is that right?
If so, is that what we want, or would we rather want to fail the pod when even a single container is crashed / exited unsuccessfully? Not sure how Kubernetes handles it tbh..

Reply:
The [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) defines the pod phase "Running" as "The Pod has been bound to a node, and all of the containers have been created. At least one container is still running, or is in the process of starting or restarting." The pod phase "Failed" is defined as "All containers in the Pod have terminated, and at least one container has terminated in failure. That is, the container either exited with non-zero status or was terminated by the system." It is not mentioned that the state of one container can affect other containers.

I would not terminate the pod if one container failed.

It must also be considered that this condition is only relevant for jobs which can terminate. Services are restarted by systemd if they terminate and are never removed from `running_containers`.

Reply:
Sounds like we adhere to K8s defaults then, which is fine for me.